src/app/zkapps_examples/tokens/zkapps_tokens.ml

```diff
@@ -691,8 +691,14 @@ module P = struct
   let verification_key =
     Lazy.bind p_module ~f:(fun (module P : Proof_intf) -> P.verification_key)
 
+  (* this is annoying *)
```

Comment:
Agreed

Reply:
with "relaxed interface", do you mean anything else than writing out the full type in the mli? I just removed the comments for now 😅

Reply:
I think it means a smaller one that does not require the annoying elements, which you could include into the expanded version

Reply:
Still don't get it. For reference the current interface just reuses an existing one:
```ocaml
module P :
  Pickles.Proof_intf
    with type statement = Zkapp_statement.t
     and type t = (Nat.N2.n, Nat.N2.n) Pickles.Proof.t
```

what would be the most efficient way to rewrite this so that `P` doesn't include `verification_key_promise` and `id_promise`

Reply:
> I think it means a smaller one that does not require the annoying elements, which you could include into the expanded version

Ok after re-reading this I get the proposal. Create a second, smaller intf, used here and also included in the larger one. Now I don't think it's worth it 😅

Reply:
I think it should at least be a follow-up issue. Your comment does mean something is not right architecture-wise.

Reply:
```